### DISCUSSION TITLE
GPU and Fortran interface ?

### DISCUSSION THREAD

**Discussion by avergnaud002** [AMReX-Codes/amrex#3259] [Q&A] [answered]
Hello,

In AMReX documentation it is stated that : _"The Fortran interface of AMReX does not currently have GPU support. AMReX recommends porting Fortran code to C++ when coding for GPUs."_
Is it still the case, or is there a "simple" way to use AMReX with GPU for codes written in fortran ?

Thanks
Alban

**Comment by WeiqunZhang** [comment, marked as answer]
Yes, this is still the case. If you call `amrex_multifab_add` from Fortran, it will run on GPU. But if you have your own functions, you can write it in C++ and provides a Fortran interface, or you can use OpenMP or OpenACC or CUDA Fortran etc.

**Reply by avergnaud002** [reply]
Ok thank you for your answer.
